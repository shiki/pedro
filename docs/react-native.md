# React Native

## Libraries

* https://reactnavigation.org/