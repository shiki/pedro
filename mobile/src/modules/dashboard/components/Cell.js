import React, { PureComponent } from 'react'
import { View, StyleSheet } from 'react-native'
import PropTypes from 'prop-types'
import { BigNumber } from 'bignumber.js'

import CellBorder from './CellBorder'
import CellDetail from './CellDetail'
import Text from '../../../components/Text'
import { Stock } from '../../../models'

import { toDisplayFormat } from '../../../utils/number'

export default class Cell extends PureComponent {
  static propTypes = {
    stock: PropTypes.instanceOf(Stock).isRequired,
    price: PropTypes.instanceOf(BigNumber).isRequired,
    operator: PropTypes.string.isRequired
  }

  render() {
    const { stock, price, operator } = this.props

    const progress = new BigNumber(1).sub(
      stock.price
        .sub(price)
        .absoluteValue()
        .dividedBy(price)
    )

    return (
      <View style={styles.container}>
        <View style={styles.topContainer}>
          <View style={styles.stockContainer}>
            <Text style={styles.stockSymbol}>
              {stock.symbol}
              <Text style={styles.stockPrice}>&nbsp;{toDisplayFormat(stock.price)}</Text>
            </Text>
            <Text style={styles.stockName}>{stock.name}</Text>
          </View>
          <Text style={styles.operator}>{operator}</Text>
          <Text style={styles.price}>{toDisplayFormat(price)}</Text>
        </View>
        <CellDetail stock={stock} />
        <CellBorder progress={progress} />
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    minHeight: 76,
    alignItems: 'stretch'
  },
  topContainer: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
    padding: 11
  },
  stockContainer: {
    flex: 6 / 10
  },
  stockSymbol: {
    fontSize: 20,
    color: '#555555'
  },
  stockPrice: {
    //
  },
  stockName: {
    fontSize: 10,
    color: '#BDBDBD'
  },
  operator: {
    flex: 1 / 10,
    fontSize: 20,
    fontFamily: 'Courier New',
    marginTop: 2,
    marginLeft: 10,
    marginRight: 10,
    textAlign: 'right'
  },
  price: {
    flex: 3 / 10,
    fontSize: 20,
    color: '#666666',
    textAlign: 'right'
  }
})
