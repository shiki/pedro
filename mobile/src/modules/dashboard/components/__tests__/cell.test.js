import 'react-native'
import React from 'react'

import renderer from 'react-test-renderer'

import moment from 'moment'
import { BigNumber } from 'bignumber.js'

import Cell from '../Cell'
// import Stock from '../../../../src/services/db/Stock'

it('renders correctly', () => {
  const data = {
    stock: { symbol: 'MER', name: 'Meralco', as_of: moment(), price: new BigNumber(100), percent_change: new BigNumber(0.931), updated_at: moment() },
    operator: '>',
    price: new BigNumber(100.991)
  }
  const tree = renderer.create(<Cell data={data} />).toJSON()
  expect(tree).toMatchSnapshot()
})
