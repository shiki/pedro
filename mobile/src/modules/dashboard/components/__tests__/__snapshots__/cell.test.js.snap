// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "flex": 1,
      "minHeight": 76,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "padding": 11,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 0.6,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#999999",
              "fontFamily": "Helvetica Neue",
              "fontSize": 14,
            },
            Object {
              "color": "#555555",
              "fontSize": 20,
            },
          ]
        }
      >
        MER
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#999999",
                "fontFamily": "Helvetica Neue",
                "fontSize": 14,
              },
              Object {},
            ]
          }
        >
           
          100.00
        </Text>
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#999999",
              "fontFamily": "Helvetica Neue",
              "fontSize": 14,
            },
            Object {
              "color": "#BDBDBD",
              "fontSize": 10,
            },
          ]
        }
      >
        Meralco
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#999999",
            "fontFamily": "Helvetica Neue",
            "fontSize": 14,
          },
          Object {
            "flex": 0.1,
            "fontFamily": "Courier New",
            "fontSize": 20,
            "marginLeft": 10,
            "marginRight": 10,
            "marginTop": 2,
            "textAlign": "right",
          },
        ]
      }
    >
      &gt;
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#999999",
            "fontFamily": "Helvetica Neue",
            "fontSize": 14,
          },
          Object {
            "color": "#666666",
            "flex": 0.3,
            "fontSize": 20,
            "textAlign": "right",
          },
        ]
      }
    >
      100.99
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "gray",
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "backgroundColor": "#F5F5F5",
          "padding": 11,
          "width": 78,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#999999",
              "fontFamily": "Helvetica Neue",
              "fontSize": 14,
            },
            Object {
              "fontSize": 10,
              "textAlign": "right",
            },
          ]
        }
      >
        Last price
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#999999",
              "fontFamily": "Helvetica Neue",
              "fontSize": 14,
            },
            Object {
              "fontSize": 12,
              "marginTop": 5,
              "textAlign": "right",
            },
          ]
        }
      >
        100.00
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#999999",
              "fontFamily": "Helvetica Neue",
              "fontSize": 14,
            },
            Object {
              "fontSize": 10,
              "textAlign": "right",
            },
            Object {
              "marginTop": 11,
            },
          ]
        }
      >
        Change
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#999999",
              "fontFamily": "Helvetica Neue",
              "fontSize": 14,
            },
            Object {
              "color": "#E57373",
            },
            Object {
              "fontSize": 12,
              "marginTop": 5,
              "textAlign": "right",
            },
          ]
        }
      >
        -32.00
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#999999",
                "fontFamily": "Helvetica Neue",
                "fontSize": 14,
              },
              Object {},
            ]
          }
        >
          

        </Text>
        -0.87%
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#FAFAFA",
          "flex": 1,
          "justifyContent": "space-between",
        }
      }
    >
      <View
        style={
          Object {
            "padding": 11,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#999999",
                "fontFamily": "Helvetica Neue",
                "fontSize": 14,
              },
              Object {
                "fontSize": 12,
              },
            ]
          }
        >
          The price went above 1,377.81 yesterday, 3:25 PM.
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "marginTop": 12,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Object {
                  "color": "#BDBDBD",
                  "fontSize": 18,
                },
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#999999",
                  "fontFamily": "Helvetica Neue",
                  "fontSize": 14,
                },
                Object {
                  "color": "#BDBDBD",
                  "fontSize": 10,
                  "marginLeft": 4,
                },
              ]
            }
          >
            Reset or change the target price to be notified again.
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-end",
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Object {
                  "color": "#757575",
                  "fontSize": 20,
                  "height": 36,
                  "lineHeight": 36,
                  "textAlign": "center",
                  "width": 36,
                },
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Object {
                  "color": "#757575",
                  "fontSize": 20,
                  "height": 36,
                  "lineHeight": 36,
                  "textAlign": "center",
                  "width": 36,
                },
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Object {
                  "color": "#757575",
                  "fontSize": 20,
                  "height": 36,
                  "lineHeight": 36,
                  "textAlign": "center",
                  "width": 36,
                },
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#EDEDED",
        "height": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#54C242",
          "flex": 1,
        }
      }
      width="99.01872444079174%"
    />
  </View>
</View>
`;
